src/chainstate/stacks/index/trie_sql.rs

```diff
@@ -171,13 +183,21 @@ pub fn write_trie_blob_to_mined<T: MarfTrieId>(
     block_hash: &T,
     data: &[u8],
 ) -> Result<u32, Error> {
-    let args: &[&dyn ToSql] = &[block_hash, &data];
-    let mut s =
-        conn.prepare("INSERT OR REPLACE INTO mined_blocks (block_hash, data) VALUES (?, ?)")?;
-    let block_id = s
-        .insert(args)?
-        .try_into()
-        .expect("EXHAUSTION: MARF cannot track more than 2**31 - 1 blocks");
+    if let Ok(block_id) = get_mined_block_identifier(conn, block_hash) {
```

Comment:
Why not `INSERT OR REPLACE`?

Reply:
`REPLACE` allocates a new rowid, even if the row already exists.